internal/kubex/config.go

```diff
@@ -35,53 +37,67 @@ type ConfigWithMeta struct {
 //
 // code inspired by sigs.k8s.io/controller-runtime@v0.13.1/pkg/client/config/config.go
 func LoadRestConfigWithMetaInformation() (*ConfigWithMeta, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
 	//  1. --kubeconfig flag
 	if kubeconfig != "" {
-		c := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfig}, nil)
-		return transform(c)
-	}
-
-	// 2. KUBECONFIG environment variable pointing at a file
-	kubeconfigPath := os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
-	if len(kubeconfigPath) == 0 {
-		if c, err := rest.InClusterConfig(); err == nil {
-			return &ConfigWithMeta{
-				K8s:            c,
-				CurrentContext: "In cluster",
-			}, nil
+		loadingRules.ExplicitPath = kubeconfig
+	} else {
+		// 2. KUBECONFIG environment variable pointing at a file
+		kubeconfigPath := os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
+		if len(kubeconfigPath) == 0 {
+			// 3. In-cluster config if running in cluster
+			if c, err := rest.InClusterConfig(); err == nil {
+				return &ConfigWithMeta{
+					K8s:            c,
+					CurrentContext: "In cluster",
+				}, nil
+			}
+		} else {
+			loadingRules.ExplicitPath = kubeconfigPath
+			// 4. $HOME/.kube/config if exists
+			// 5. user.HomeDir/.kube/config if exists
+			//
+			// NOTE: For default config file locations, upstream only checks
+			// $HOME for the user's home directory, but we can also try
+			// os/user.HomeDir when $HOME is unset.
+			if _, ok := os.LookupEnv("HOME"); !ok {
+				u, err := user.Current()
+				if err != nil {
+					return nil, fmt.Errorf("could not get current user: %w", err)
+				}
+				loadingRules.Precedence = append(loadingRules.Precedence, filepath.Join(u.HomeDir, clientcmd.RecommendedHomeDir, clientcmd.RecommendedFileName))
+			}
 		}
 	}
 
-	// 3. In-cluster config if running in cluster
-	// 4. $HOME/.kube/config if exists
-	// 5. user.HomeDir/.kube/config if exists
-	//
-	// NOTE: For default config file locations, upstream only checks
-	// $HOME for the user's home directory, but we can also try
-	// os/user.HomeDir when $HOME is unset.
-	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
-	if _, ok := os.LookupEnv("HOME"); !ok {
-		u, err := user.Current()
-		if err != nil {
-			return nil, fmt.Errorf("could not get current user: %w", err)
-		}
-		loadingRules.Precedence = append(loadingRules.Precedence, filepath.Join(u.HomeDir, clientcmd.RecommendedHomeDir, clientcmd.RecommendedFileName))
+	//  1. --kubecontext flag
+	if kubecontext == "" {
+		// 2. KUBECONTEXT env
+		kubecontext = os.Getenv("KUBECONTEXT")
 	}
 
-	return transform(clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, nil))
+	configOverrides := &clientcmd.ConfigOverrides{CurrentContext: kubecontext}
+	return transform(clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides))
 }
```

Comment:
What do you think about extracting the kubecontext logic to a separate function? E.g. something like

```go
	configOverrides := &clientcmd.ConfigOverrides{CurrentContext: loadKubecontext()}
	return transform(clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides))
}

func loadKubecontext() string {
	//  1. --kubecontext flag
	if kubecontext != "" {
		return kubecontext
	}

	// 2. KUBECONTEXT env
	return os.Getenv("KUBECONTEXT")
}
```

Reply:
Sounds reasonable to me.